src/cgtop/cgtop.c

```diff
@@ -835,7 +836,15 @@ static int parse_argv(int argc, char *argv[]) {
                         assert_not_reached("Unhandled option");
                 }
 
-        if (optind < argc) {
+        if (optind == argc-1) {
+                if (arg_machine) {
+                        log_error("Specify a cgroup path together with the -M option is not allowed");
+                        return -EINVAL;
+                }
+                arg_root = strdup(argv[optind]);
+                if (!arg_root)
+                        return log_oom();
```

Comment:
Hmm, I am pretty sure arg_root should point to argv[optind] directly here, as that memory is stable and stays around. However, get_cgroup_root() should really do a copy then, so that it always returns malloc()ed memory, and not sometimes global variables... (see below)